tests/types/test_lazy_types_future_annotations.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import textwrap
+from typing_extensions import Annotated
+
+import strawberry
+
+
+def test_optional_lazy_type_using_or_operator():
+    from tests.schema.test_lazy.type_a import TypeA
+
+    global SomeType, AnotherType
+
+    try:
+
+        @strawberry.type
+        class SomeType:
+            foo: (
+                Annotated[TypeA, strawberry.lazy("tests.schema.test_lazy.type_a")]
+                | None
+            )
+
+        @strawberry.type
+        class AnotherType:
+            foo: TypeA | None = None
+
+        @strawberry.type
+        class Query:
+            some_type: SomeType
+            another_type: AnotherType
+
+        schema = strawberry.Schema(query=Query)
+        expected = """\
+        type AnotherType {
+          foo: TypeA
+        }
+
+        type Query {
+          someType: SomeType!
+          anotherType: AnotherType!
+        }
+
+        type SomeType {
+          foo: TypeA
+        }
+
+        type TypeA {
+          listOfB: [TypeB!]
+          typeB: TypeB!
+        }
+
+        type TypeB {
+          typeA: TypeA!
+          typeAList: [TypeA!]!
+          typeCList: [TypeC!]!
+        }
+
+        type TypeC {
+          name: String!
+        }
+        """
+        assert str(schema).strip() == textwrap.dedent(expected).strip()
+    finally:
+        del SomeType, AnotherType
```

Comment:
Just out of curiosity: why do these types have to be globals and need to be cleaned up like this at the end?

Reply:
They need to be global because the lazy importer will try to import it from the module, and in here we are defining those inside the function.

Another option would be to define the types themselves at the module level, but then it becomes difficult to change them for other tests :(

The cleanup at the end is just to make sure that the global namespace doesn't get polluted and possibly affects other tests

Reply:
Thanks for the explanation Thiago, this makes a lot of sense.